### PR TITLE
Add FXIOS-4575 [v105] Add notification tests for recently saved

### PR DESCRIPTION
**Pull request by OrlaM**
Updated NotificationCenter to conform to a protocol so I can mock it in tests
Added some more tests for the recently saved adaptor

### CODE REVIEW

**Review by electricRGB — APPROVED** [review]

File: Client/Protocols/Notifiable.swift [hunk 1]
```diff
@@ -4,8 +4,24 @@
...
+    func removeObserver(_ observer: Any)
+}
+
+extension NotificationCenter: NotificationProtocol {
+    func post(name: NSNotification.Name) {
+        self.post(name: name, object: nil)
+    }
+//    func post(name: NSNotification.Name, object anObject: Any?) {
+//        super.post(name: name, object: anObject)
+//    }
```
Comment (lmarceau) [inline Client/Protocols/Notifiable.swift:20]: Commented code?

File: Client/Protocols/Notifiable.swift [hunk 2]
```diff
@@ -4,8 +4,24 @@
 
 import Foundation
 
+@objc protocol NotificationProtocol {
+    func post(name: NSNotification.Name)
+    //func post(name: NSNotification.Name, object anObject: Any?)
```
Comment (lmarceau) [inline Client/Protocols/Notifiable.swift:9]: Commented code too here?

File: Tests/ClientTests/Frontend/Home/RecentlySaved/RecentlySavedDataAdaptorTests.swift [hunk 3]
```diff
@@ -13,15 +13,15 @@ class RecentlySavedDataAdaptorTests: XCTestCase {
...
+    var mockNotificationCenter: MockNotificationCenter!
     var mockDelegate: RecentlySavedDelegateMock?
 
     override func setUp() {
         super.setUp()
         mockSiteImageHelper = SiteImageHelperMock()
         mockReadingList = ReadingListMock()
         mockBookmarksHandler = BookmarksHandlerMock()
-        spyNotificationCenter = SpyNotificationCenter()
+        mockNotificationCenter = MockNotificationCenter()
```
Comment (lmarceau) [inline Tests/ClientTests/Frontend/Home/RecentlySaved/RecentlySavedDataAdaptorTests.swift:24]: Ah nice! That was my next question if we should replace that Spy haha. Thanks!

Comment (OrlaM) [reply Tests/ClientTests/Frontend/Home/RecentlySaved/RecentlySavedDataAdaptorTests.swift:24]: I think the spy is still fine to use for testing classes that post notifications but it's more difficult to use for testing listeners without adding async code to the tests which is why I couldn't really make use of it here.

**Review by lmarceau — APPROVED** [review]

**Comment by OrlaM** [comment]
Tests are passing even though the report shows a fail.
image

**Pull request merged**